packages/cli/lib/services/response-saver.service.ts

```diff
@@ -60,35 +83,99 @@ export function onAxiosRequestFulfilled({
         saveResponse<Pick<Connection, 'metadata' | 'connection_config'>>({
             directoryName,
             data: { metadata: connection.metadata as Metadata, connection_config: connection.connection_config },
-            customFilePath: 'mocks/nango/getConnection.json',
-            concatenateIfExists: false
+            customFilePath: 'mocks/nango/getConnection.json'
         });
 
         saveResponse<Metadata>({
             directoryName,
             data: connection.metadata as Metadata,
-            customFilePath: 'mocks/nango/getMetadata.json',
-            concatenateIfExists: false
+            customFilePath: 'mocks/nango/getMetadata.json'
         });
 
         return response;
     }
 
-    const [pathname, params] = response.request.path.split('?');
-    const strippedPath = pathname.replace('/', '');
-
-    let concatenateIfExists = false;
-
-    if (params && params.includes('page')) {
-        concatenateIfExists = true;
-    }
+    const requestIdentity = computeConfigIdentity(response.config);
 
-    saveResponse<AxiosResponse>({
+    saveResponse<CachedRequest>({
         directoryName,
-        data: response.data,
-        customFilePath: `mocks/nango/${method}/${strippedPath}/${syncName}.json`,
-        concatenateIfExists
+        data: {
+            ...requestIdentity,
+            response: response.data,
+            status: response.status,
+            headers: response.headers as Record<string, string>
+        },
+        customFilePath: `mocks/nango/${requestIdentity.method}/proxy/${requestIdentity.endpoint}/${syncName}/${requestIdentity.requestIdentityHash}.json`
     });
 
     return response;
 }
+
+function computeConfigIdentity(config: AxiosRequestConfig): ConfigIdentity {
+    const method = config.method?.toLowerCase() || 'get';
+    const params = sortEntries(Object.entries(config.params || {}));
+
+    const url = new URL(config.url!);
+    const endpoint = url.pathname.replace(/^\/proxy\//, '');
+
+    const dataIdentity = computeDataIdentity(config);
+
+    let headers: [string, string][] = [];
+    if (config.headers !== undefined) {
+        const filteredHeaders = Object.entries(config.headers).filter(([key]) => !FILTER_HEADERS.includes(key.toLowerCase()));
+        sortEntries(filteredHeaders);
+        headers = filteredHeaders;
+    }
+
+    // order is important to the request hash
+    const requestIdentity = {
```

Comment:
Not requesting but you would spot errors before at the right place with this
```suggestion
    const requestIdentity: RequestIdentity = {
```